django_diplom/catalog/views.py

```diff
@@ -0,0 +1,170 @@
+from django.core.handlers.wsgi import WSGIRequest
+from django.db.models import Count
+from django.http import HttpResponse, Http404
+from django.shortcuts import render, redirect
+
+from django.views.generic import TemplateView
+
+from cart.models import CartProduct, CartAddon
+from catalog.models import Category, Product, Addon
+
+
+class HomeTemplateView(TemplateView):
+    template_name = "catalog/home.html"
+
+    def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
+        context: Dict[str, Any] = super().get_context_data(**kwargs)
+        context['categories'] = Category.objects.annotate(
+            total=Count('product', distinct=True) + Count('addon', distinct=True))
+
+        return context
+
+
+@login_required
+def home(request):
+    cat = Category.objects.annotate(Count('product'))
+    return render(
+        request,
+        'catalog/home.html',
+        {
+            "categories": cat,
+
+        }
+    )
+
+
+class MenTemplate(TemplateView):
+    template_name = "catalog/Men.html"
+
+    def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
+        context: Dict[str, Any] = super().get_context_data(**kwargs)
+        context['categories'] = Category.objects.annotate(total=Count('product', distinct=True)).filter(gender='male')
+
+        return context
+
+
+def men(request):
+    cat = Category.objects.annotate(Count('product'))
+    return render(
+        request,
+        'catalog/Men.html',
+        {
+            "categories": cat,
+
+        }
+    )
+
+
+def kids(request):
+    cat = Category.objects.filter(gender='kids')
+    return render(
+        request,
+        'catalog/Kids.html',
+        {
+            "categories": cat,
+
+        }
+    )
+
+
+def women(request):
+    cat = Category.objects.filter(gender='female')
+    return render(
+        request,
+        'catalog/Women.html',
+        {
+            "categories": cat,
+        }
+    )
+
+
+def accessory(request):
+    cat = Category.objects.filter(gender='accessory')
+    return render(
+        request,
+        'catalog/Women.html',
+        {
+            "categories": cat,
+        }
+    )
+
+
+def addon(request):
+    addon = Category.objects.annotate(Count('addon'))
+    return addon
+
+
+def cat(request: WSGIRequest, category_slug: str) -> HttpResponse:
+    try:
+
+        category: Category = (Category.objects.prefetch_related("product_set").get(slug=category_slug))
+    except Category.DoesNotExist:
+        raise Http404
+    return render(request, 'catalog/home.html', {'category': category})
+
+
+@login_required
+def one_cat(request, category_slug):
+    cat = Category.objects.prefetch_related("product_set").prefetch_related("addon_set__product_set").filter(
+        slug=category_slug).first()
+
+    return render(
+        request,
+        'catalog/one_categories.html',
+        {"categories": cat,
+         "addons": addon(request)
+         }
+
+    )
+
+
+def prod_view(request, product_slug):
```

Comment:
```suggestion
def product_view(request, product_slug):
```
Не стоит экономить буквы)